### PR TITLE
scripts and doc re-organisation

### PR DESCRIPTION
**Pull request by jburel**
* Re-organise the documentation. The original file has now be broken in several files
* Migrate with history groovy script from training-scripts repo
* Include code snippets
* Provide python script similar to notebook

This should make it easier to include example and doc using zarr files.
The "tracking mitosis" file will be reviewed in an other PR and replace if possible with a notebook using IDR
cc @pwalczysko

### CODE REVIEW

File: docs/gettingstarted.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
+Getting started with ilastik API and OMERO
+==========================================
+
+**Description**
+---------------
+
+We will use a Python script showing how to analyze data stored in an OMERO server
+using the ilastik API. The code snippets are extracted from the Python script
+:download:`pixels_classification.py <../scripts/pixels_classification.py>`. A notebook
```
Comment (pwalczysko) [inline docs/gettingstarted.rst]: Could we maybe rename the script to say ``pixel_classification.py`` (remove the ``s``) ?

File: docs/gettingstarted.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+is also available, see `pixels_classification.ipynb <https://mybinder.org/v2/gh/ome/omero-guide-ilastik/master?filepath=notebooks>`_.
```
Comment (pwalczysko) [inline docs/gettingstarted.rst]: Also rename the notebook accordingly https://github.com/ome/omero-guide-ilastik/pull/10/files#r413807260 ?

Comment (jburel) [reply docs/gettingstarted.rst]: done

File: docs/gettingstarted.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+- How load images from a dataset using the OMERO API.
+
+- How to run ilastik using its Python API.
+
+- How to save the generated results as OMERO images.
+
+**Resources**
+-------------
+
+We will use an ilastik project created with ilastik version 1.3.3 to analyse 3D tissues
```
Comment (pwalczysko) [inline docs/gettingstarted.rst]: Confusingly, this is not considered to be a ``tissue`` and it is not in Tissue-IDR (rather Cell-IDR). Maybe

``...analyse 3D tissues data from the Image Data Resource (IDR).`` ->  ``...analyze 3D images of mouse blastocysts from the Image Data Resource (IDR)``

**Comment by pwalczysko** [comment]
I think we have to unify the order of the parts ``Setup`` and ``Resources``. 
There are both cases in the different Ilastik guide chapters, ``Setup`` before ``Resoureces`` and also ``Setup`` after ``Resources`` (e.g.  ``Getting started with ilastik API ...`` and ``Use ilastik as a Fiji plugin...`` have different order)

**Comment by jburel** [comment]
order swapped

**Comment by pwalczysko** [comment]
```
conda env update -n ilastik --file binder/environment.yml
....
UnsatisfiableError: The following specifications were found                     
to be incompatible with the existing python installation in your environment:

Specifications:

  - ilastik-meta -> python[version='>=3.7,<3.8.0a0']
  - omero-py -> python[version='2.7.*|3.5.*|3.7.*']
  - omero-py -> python[version='>=2.7,<2.8.0a0|>=3.5,<3.6.0a0|>=3.8,<3.9.0a0']
  - pip -> python=3.7

Your python: python=3.6
...
```

When following 
Screen Shot 2020-04-23 at 15 36 43

**Comment by jburel** [comment]
I copied this section from CP, this won't work since few more things need to happen

**Comment by pwalczysko** [comment]
The other fixes look good. The https://github.com/ome/omero-guide-ilastik/pull/10#issuecomment-618435274 needs to be checked. Mybinder was also failing for me now to build the notebook, but will come to that later.

**Comment by jburel** [comment]
@pwalczysko https://github.com/ome/omero-guide-ilastik/pull/10#issuecomment-618435274 should be fixed

**Comment by jburel** [comment]
I have checked on mybinder again this morning and it works
Did you use the correct repo and branch?

**Comment by pwalczysko** [comment]
> I have checked on mybinder again this morning and it works
Did you use the correct repo and branch?

Sorry, did not understand about the branches. Tried again with this repo here and the branch scripts. Binder worked fine, saved 2 probability images ot OMERO (connection failed after that)

It seems to work.

**Comment by jburel** [comment]
are you happy with the current state?
Still some work to be done on the mitosis front, but that will happen in another PR

**Comment by pwalczysko** [comment]
Sorry, still cannot find this bit in the code
The part below is very abrupt - I installed my ilastik and omero-py into the conda env as instructed. Instead of telling me what comes next, it says: ``Alternatively, do repo2docker``. And after that, only the ``repo2docker`` story line is followed. Some conclusion would be good, like "now start using the ilastik API using the other chapter in this section ?". 

Screen Shot 2020-04-24 at 14 00 53

**Comment by jburel** [comment]
I guess the "bit" you indicating is the command in ``binder/postBuild``

**Comment by pwalczysko** [comment]
Where is ``binder/postBuild`` ? - actually, sorry about the blue highlight in my screenshot, this is irrelevant - by bit I mean the whole page, including the whole section in the screenshot

**Comment by jburel** [comment]
there is a folder named ``binder`` and you will a file ``postBuild`` in the omero-guide-ilastik directory

**Comment by pwalczysko** [comment]
The new Options in the installation instructions look fine. 

Also, did the install of ilastik locally and then ran the pixel_slassification.py script from the local environment. It gives some error messages at the start, but it works as expected, the probabilities image was saved to OMERO.

Running ``repo2docker .`` now

**Comment by pwalczysko** [comment]
``repo2docker`` works as expected too. I think this gives us a nice range of use cases.
Thanks
Ready to merge fmpov.

**Comment by jburel** [comment]
Thanks @pwalczysko for the review and the feedback 
I will merge so we have the new doc updated

**Pull request merged**